src/compiler/tsc.ts

```diff
@@ -146,131 +173,176 @@ module ts {
             return sys.exit(EmitReturnStatus.CompilerOptionsErrors);
         }
 
-        if (compilerOptions.version) {
+        if (commandLine.options.version) {
             reportDiagnostic(createCompilerDiagnostic(Diagnostics.Version_0, version));
             return sys.exit(EmitReturnStatus.Succeeded);
         }
 
-        if (compilerOptions.help) {
+        if (commandLine.options.help) {
             printVersion();
             printHelp();
             return sys.exit(EmitReturnStatus.Succeeded);
         }
 
-        if (commandLine.filenames.length === 0) {
+        if (commandLine.options.project) {
+            configFilename = normalizePath(combinePaths(commandLine.options.project, "tsconfig.json"));

```

Comment:
I doesn't crash because we catch the exception when attempting to JSON.parse the file. What you'll see is an error message saying
```
error TS6050: Unable to open file 'tsconfig.json'.
```
I like this solution because it also alerts you in cases where we find the tsconfig.json file by walking the directory path but then fail because of no JSON support.